artemis-server/src/main/java/org/apache/activemq/artemis/core/server/group/impl/LocalGroupingHandler.java

```diff
@@ -322,11 +322,19 @@ public synchronized void start() throws Exception {
       if (started)
          return;
 
-      if (expectedBindings == null) {
-         // just in case the component is restarted
-         expectedBindings = new LinkedList<>();
+      try
+      {
+         lock.lock();
```

Comment:
The attempt to lock is normally placed outside the try as you don't want the finally to attempt to unlock a lock if the caller fails to acquire the lock in the first place.